pkg/auth/auth.go

```diff
@@ -90,3 +109,26 @@ func init() {
 		panic(err)
 	}
 }
+
+// InitReporter initialize the reporter with given reporter
+func InitReporter(reporterInstance *metrics.Reporter) {
+	reporter = reporterInstance
+}
+
+// recordError records the error in appropriate metric
+func recordError(operation string) {
```

Comment:
Can we simplify this by having one single call in the metrics package and all other modules just call in with the counters required in a follow up PR.

Reply:
metrics package has 'ReportOperation','Report' to record given measurements.
Internally same is used by recordError.It has been abstracted by recordError to avoid code cluttering and to refer particular measurement at single place for any future changes.

Following are the options we can think of .
1. Keep it as-is
2. Move ReportCloudProviderOperations() to metrics package
3. Use reporter.ReportOperation with measurement name instead of recordError

@kkmsft  - Let me know which options sounds good ?

`func recordError(operation string) {
	if reporter != nil {
		reporter.ReportOperation(
			operation,
			metrics.CloudProviderOperationsErrorsCountM.M(1))
	}
}`